components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/SalesforceHttpClient.java

```diff
@@ -113,6 +113,12 @@ protected void doStart() throws Exception {
         super.doStart();
     }
 
+    @Override
+    protected void doStop() throws Exception {
+        workerPool.shutdown();
```

Comment:
Use executorServiceManager to shutdown the pool, and also do a if != null check (as camel can stop a service if it failed to start so the pool can potentially be null)

Reply:
@davsclaus ok, now uses ExecutorServiceManager, if available. It won't be available if SalesforceHttpClient is instantiated outside of Camel, as is the case for the tooling that uses SalesforceHttpClient.